test/object-store/sync/client_reset.cpp

```diff
@@ -232,27 +232,40 @@ TEST_CASE("sync: pending client resets are cleared when downloads are complete",
 
         FAIL(util::format("got error from server: %1", err.status));
     };
-
-    auto realm = Realm::get_shared_realm(realm_config);
-    auto obj_id = ObjectId::gen();
     {
-        realm->begin_transaction();
-        CppContext c(realm);
-        Object::create(
-            c, realm, "object",
-            std::any(AnyDict{{"_id", obj_id}, {"value", int64_t(5)}, {partition.property_name, partition.value}}));
-        realm->commit_transaction();
-        wait_for_upload(*realm);
+        auto realm = Realm::get_shared_realm(realm_config);
+        auto obj_id = ObjectId::gen();
+        {
+            realm->begin_transaction();
+            CppContext c(realm);
+            Object::create(c, realm, "object",
+                           std::any(AnyDict{
+                               {"_id", obj_id}, {"value", int64_t(5)}, {partition.property_name, partition.value}}));
+            realm->commit_transaction();
+            wait_for_upload(*realm);
+        }
+        wait_for_download(*realm, std::chrono::minutes(10));
+        reset_utils::trigger_client_reset(test_app_session.app_session(), realm);
+    }
+    {
+        // Reconnect and wait for the client reset to complete
+        auto [future, after_reset] = reset_utils::make_client_reset_handler();
+        realm_config.sync_config->notify_after_client_reset = after_reset;
+        auto realm = Realm::get_shared_realm(realm_config);
+        wait_for_download(*realm, std::chrono::minutes(10));
+        REQUIRE(future.is_ready());
```

Comment:
What are we checking here?

Reply:
Its supposed to verify that the client reset is complete (e.g. `notify_after_client_reset` callback has already been called and the future is ready) by the time the `wait_for_download()` returns after the client reset is performed.

I am assuming that was the original intent for this test, but the original test wasn't actually performing any client resets.

Reply:
it looks like the original test was calling trigger_client_reset, did that not work for some reason? or was it just not checking that a client reset actually happened?

Reply:
`trigger_client_reset()` just invalidates the file ident and the client needs to reconnect before it will receive the client reset error.

Reply:
so this way we're actually triggering a client reset because we close the realm each time by putting the `realm` in separate scopes?

Reply:
Yes - since it's using the same config, it should be reusing the same realm file/file ident.
Would you prefer me to use `pause()`/`resume()` instead?